test/test_graph.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
```

Comment:
nit, lets call this `graph_test.py`, I think executable tests should all end in `_test.py`

Reply:
whatever the naming convention is lets stick to it for all tests and make sure that `test_base.py` does not look like a runnable test